shell/platform/windows/flutter_windows.cc

```diff
@@ -0,0 +1,242 @@
+  // flags the first item is treated as the executable and ignored. Add a dummy
+  // value so that all provided arguments are used.
+  std::vector<const char*> argv = {"placeholder"};
+  if (arguments_count > 0) {
+    argv.insert(argv.end(), &arguments[0], &arguments[arguments_count]);
+  }
+
+  window->CreateRenderSurface();
+
+  FlutterRendererConfig config = {};
+
+  // Provide the necessary callbacks for rendering within a win32 window.
+  config.type = kOpenGL;
+  config.open_gl.struct_size = sizeof(config.open_gl);
+  config.open_gl.make_current = [](void* user_data) -> bool {
+    auto host = static_cast<flutter::Win32FlutterWindow*>(user_data);
+    return host->MakeCurrent();
+  };
+  config.open_gl.clear_current = [](void* user_data) -> bool {
+    auto host = static_cast<flutter::Win32FlutterWindow*>(user_data);
+    return host->ClearContext();
+  };
+  config.open_gl.present = [](void* user_data) -> bool {
+    auto host = static_cast<flutter::Win32FlutterWindow*>(user_data);
+    return host->SwapBuffers();
+  };
+  config.open_gl.fbo_callback = [](void* user_data) -> uint32_t { return 0; };
+  config.open_gl.gl_proc_resolver = [](void* user_data,
+                                       const char* what) -> void* {
+    return eglGetProcAddress(what);
+  };
+
+  FlutterProjectArgs args = {};
+  args.struct_size = sizeof(FlutterProjectArgs);
+  args.assets_path = assets_path;
+  args.icu_data_path = icu_data_path;
+  args.command_line_argc = static_cast<int>(argv.size());
+  args.command_line_argv = &argv[0];
+  args.platform_message_callback =
+      [](const FlutterPlatformMessage* engine_message,
+         void* user_data) -> void {
+    auto window = reinterpret_cast<flutter::Win32FlutterWindow*>(user_data);
+    return window->HandlePlatformMessage(engine_message);
+  };
+
+  // args.custom_task_runners = custom_task_runners; TODO
+
+  FLUTTER_API_SYMBOL(FlutterEngine) engine = nullptr;
+  auto result =
+      FlutterEngineRun(FLUTTER_ENGINE_VERSION, &config, &args, window, &engine);
+  if (result != kSuccess || engine == nullptr) {
+    std::cerr << "Failed to start Flutter engine: error " << result
+              << std::endl;
+    return nullptr;
+  }
+
+  return engine;
+}
+
+bool FlutterDesktopInit() {
+  return true;
+}
+
+void FlutterDesktopTerminate() {}
+
+FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
+    int initial_width,
+    int initial_height,
+    const char* title,
+    const char* assets_path,
+    const char* icu_data_path,
+    const char** arguments,
+    size_t argument_count) {
+  auto win = std::make_unique<flutter::Win32FlutterWindow>(
+      title, 10, 10, initial_width, initial_height);
+
+  flutter::Win32FlutterWindow* window = win.get();
+
+  auto engine = RunFlutterEngine(window, assets_path, icu_data_path, arguments,
+                                 argument_count);
+
+  if (engine == nullptr) {
+    return nullptr;
+  }
+
+  auto state = window->SetState(engine);
```

Comment:
s/auto/FlutterDesktopWindowControllerRef/ here; the type is non-obvious.

Reply:
Fixed in next commit